### PR TITLE
migrate domain-server basic auth to use SHA256 hash

### PR DESCRIPTION
**Pull request by birarda**
- store the password for HTTP basic auth in domain-server as SHA256 hash

### CODE REVIEW

**Comment by birarda** [comment]
Things to test:

(1) migration of a password from previous settings version that was in plaintext
- run a domain-server from before this build and ensure you have an HTTP username/password set
- run this domain-server and ensure that it says "Migrating plaintext password to base64 password in domain-server settings" and the the password in your config.json file is now in base64

(2) setting of a new password from domain-server settings
- run a domain-server from this build and change the HTTP password
- ensure it is Base64 encoded in config.json

In both cases above, make sure you can log in with the password once you see it in config.json in Base64.

**Comment by birarda** [comment]
Fix for https://app.asana.com/0/26225263936266/107569230919812

**Comment by hifi-gustavo** [comment]
Build console output for commit: 99d0e4197338b0c7f3be35e1f85e8b37dcd01ba6 available at: https://gist.github.com/be40e126f7a91ebb23f4a399f7d8b354<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7577.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7577.dmg

File: domain-server/src/DomainServerSettingsManager.cpp [hunk 1]
```diff
@@ -165,6 +167,28 @@ void DomainServerSettingsManager::setupConfigMap(const QStringList& argumentList
...
+            static const QString BASIC_AUTH_PASSWORD_KEY_PATH { "security.http_password" };
+
+            QVariant* passwordVariant = valueForKeyPath(_configMap.getUserConfig(), BASIC_AUTH_PASSWORD_KEY_PATH);
+
+            if (passwordVariant && passwordVariant->canConvert(QMetaType::QString)) {
+                QString plaintextPassword = passwordVariant->toString();
+
+                qDebug() << "Migrating plaintext password to base64 password in domain-server settings.";
+
+                *passwordVariant = plaintextPassword.toUtf8().toBase64();
```
Comment (huffman) [inline domain-server/src/DomainServerSettingsManager.cpp]: Why store base64 instead of hash? base64 is reversible

Comment (birarda) [inline domain-server/src/DomainServerSettingsManager.cpp]: good point, the HTTP basic auth sends it as Base64 so I was just going to make it match but I don't see any reason why we wouldn't make it irreversible in the settings.

Comment (huffman) [inline domain-server/src/DomainServerSettingsManager.cpp]: Ah, that makes sense. Agreed, may as well make it irreversible in the settings.

**Comment by birarda** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: a0edbac486c090106b0e38dfd56a60095895f17a available at: https://gist.github.com/be40e126f7a91ebb23f4a399f7d8b354<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7577.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7577.dmg

**Comment by OmegaHeron** [comment]
Compiling this for testing on my two remote Linux servers now.

**Comment by OmegaHeron** [comment]
Tested Linux - locally stored plain text password updated to hashed version.  Able to login.  I run nginx as a proxy in front of stack to force SSL so it's not somewhat better secured from someone on local machine snooping should they gain access and via network snooping.  QA Pass

When are we going to talk about using SSL on domain-server?  The code seems to be there, but it was far easier for me to just proxy it via nginx already running on same machines so I never bothered digging into that.

**Pull request merged**